syncd/syncd_reinit.cpp

```diff
@@ -610,6 +617,174 @@ void helperCheckVlanId()
     g_redisClient->hset(strKey, "NULL", "NULL");
 }
 
+sai_uint32_t saiGetPortNumberOfQueues(sai_object_id_t portId)
+{
+    SWSS_LOG_ENTER();
+
+    sai_attribute_t attr;
+
+    attr.id = SAI_PORT_ATTR_QOS_NUMBER_OF_QUEUES;
+
+    sai_status_t status = sai_port_api->get_port_attribute(portId, 1, &attr);
+
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("failed to get port number of queues: %d", status);
+
+        exit_and_notify(EXIT_FAILURE);
+    }
+
+    SWSS_LOG_DEBUG("port 0x%llx queues number: %u", portId, attr.value.u32);
+
+    return attr.value.u32;
+}
+
+std::vector<sai_object_id_t> saiGetPortQueues(sai_object_id_t portId)
+{
+    SWSS_LOG_ENTER();
+
+    uint32_t queueCount = saiGetPortNumberOfQueues(portId);
+
+    std::vector<sai_object_id_t> queueList;
+
+    queueList.resize(queueCount);
```

Comment:
what ? is zero ? then why there is inconsistency whit that ? as you said before how we can't have queues ?